storage_controller/src/compute_hook.rs

```diff
@@ -124,11 +138,15 @@ impl ComputeHookTenant {
         &mut self,
         tenant_shard_id: TenantShardId,
         stripe_size: ShardStripeSize,
+        preferred_az: Option<&AvailabilityZone>,
         node_id: NodeId,
     ) {
         match self {
             Self::Unsharded(unsharded_tenant) if tenant_shard_id.shard_count.count() == 1 => {
-                unsharded_tenant.node_id = node_id
+                unsharded_tenant.node_id = node_id;
+                if unsharded_tenant.preferred_az.as_ref() != preferred_az {
+                    unsharded_tenant.preferred_az = preferred_az.cloned();
+                }
```

Comment:
When I started looking at this I found that I'm not sure there's a terribly good reason for having these variants at all, rather than just using a shard-aware type everywhere.  Not sure why I wrote it this way to begin with (maybe just to make it easier to reason about?).  So let's plan on doing a larger refactor later to eliminate the two types.